### ISSUE TITLE
[JOSS Review] Test requirements

### ISSUE THREAD

**Issue by 1313e** [mpi4jax/mpi4jax#93]
This issue is part of the JOSS Review in openjournals/joss-reviews#3419.

I would like to execute the tests in `mpi4jax`, which use `pytest`, on my desktop, but there are no instructions nor requirements available for performing these tests.
Could a list of requirements be added, either as a file that can be installed with `pip install -r requirements_test.txt`, or simply adding it to the `setup()` function in the `setup.py` file?

**Comment by PhilipVinc** [comment]
@1313e thanks for your review.

@dionhaefner is it ok if we go with adding an `extras_require={'dev':[pytest, black, precommit...]}` so that if one does `pip install mpi4jax[dev]` then everything will work out of the box?

PyTest is already set up to run the tests by simply invoking `pytest` so we might need to write in the documentation somewhere that you are supposed to invoke `pytest && mpirun -np 2 pytest`

**Comment by dionhaefner** [comment]
Yes, sounds good. So if you take care of the extras, I can expand the README a bit with a mini developer & contribution guide.

**Comment by PhilipVinc** [comment]
readme or a section in the documentation?

**Comment by dionhaefner** [comment]
Sure, I'll add it to the docs then.

**Comment by dionhaefner** [comment]
How's this?
https://github.com/mpi4jax/mpi4jax#community-guidelines
https://mpi4jax.readthedocs.io/en/latest/developers.html

**Comment by 1313e** [comment]
That looks good to me.

**Comment by 1313e** [comment]
However, it seems that `pip install -e .[dev]` does not install anything extra, as `pip` does not understand that command:
```WARNING: mpi4jax 0.3.1+2.g64f6408 does not provide the extra 'dev'```.

**Comment by dionhaefner** [comment]
Try again, please.

**Comment by 1313e** [comment]
Might be a good idea to install the package using the described way from the online documentation in GitHub Actions, to make sure that there is a minimal difference between the user way and automated way of testing.
